EpikCord/__init__.py

```diff
@@ -1156,6 +1113,151 @@ async def command_error(
+        return before,after
+        
+    async def auto_moderation_rule_delete(self, data:dict)->Optional[AutoModerationRule]:
+        rule_guild_id = data["guild_id"]
+        previous:AutoModerationRule = self.auto_moderation_cache.get(rule_guild_id)
+        return previous
+
+    async def auto_moderation_action_execution(self, data:dict) -> AutoModerationExecutionEvent:
+        return AutoModerationExecutionEvent(data)    
+
+    # Guild Events
+    async def guild_create(self, data) -> Union[UnavailableGuild, Guild]:
+        guild = (
+            UnavailableGuild(data)
+            if data.get("unavailable") is True
+            else Guild(self, data)
+            if data.get("unavailable") is False
+            else None
+        )
+
+        if not guild:
+            return
+
+        self.guilds.add_to_cache(guild.id, guild)
+
+        if data.get("unavailable") is None:
+            return  # Bot was removed
+
+        for channel in data["channels"]:
+            self.channels.add_to_cache(
+                data["id"], self.utils.channel_from_type(channel)
+            )
+
+        for thread in data["threads"]:
+            self.channels.add_to_cache(data["id"], self.utils.channel_from_type(thread))
+
+        return guild
+
+        # TODO: Add other attributes to cache
+    async def guild_update(self, data) -> tuple[Optional[Guild], Guild]:
+        after:Guild = Guild(self, data)
+        before:Optional[Guild] = self.guilds.fetch(after.id)
+        return before ,after
+    
+    async def guild_delete(self, data: dict) -> Optional[Guild]:
+        return self.guilds.get(data["id"])
+
+    async def guild_ban_add(self, data:dict) -> tuple[Optional[Guild], User]:
+        return self.guilds.fetch(data["guild_id"]), User(self, data["user"])
+
+    async def guild_ban_remove(self, data:dict)-> tuple[Optional[Guild], User]:
+        return self.guilds.fetch(data["guild_id"]), User(self, data["user"])
+    
+    async def guild_emojis_update(self,data:dict) -> tuple[Optional[Guild], List[Emoji]]:
+        emojis = [Emoji(self,emoji, data["guild_id"]) for emoji in data["emojis"]]
+        return self.guilds.fetch(data["guild_id"]), emojis
+
+    async def guild_stickers_update(self,data:dict) -> tuple[Optional[Guild], List[Sticker]]:
+        stickers = [Sticker(sticker) for sticker in data["stickers"]]
+        return self.guilds.fetch(data["guild_id"]), stickers 
+
+    async def guild_integrations_update(self,data:dict) -> Optional[Guild]:
+        return self.guilds.fetch(data["guild_id"])
+
+    async def guild_member_add(self,data:dict) -> tuple[Optional[Guild], GuildMember]:
+        return self.guilds.fetch(data["guild_id"]), GuildMember(self, data)    
+
+    async def guild_member_update(self, data) -> tuple[Optional[GuildMember],GuildMember]:
+        guild_member = GuildMember(self, data)
+        return self.members.fetch(data["id"]), guild_member
```

Comment:
Because the event will never return invalid data, the guild_id will always be present. Use `get` otherwise they're the same thing and you wasted an API cal.